src/entry/data-provider/index.ts

```diff
@@ -31,6 +35,32 @@ export const dataProvider = async (request: DataProviderPayload): Promise<DataPr
 
   const trackingBranch = await getTrackingBranchName(groupToken, projectId, defaultBranch);
 
+  const backfillData: BackfillData = {
+    builds: [],
+    deployments: [],
+    metrics: {
+      mrCycleTime: 0,
+      openMergeRequestsCount: 0,
+    },
+  };
+
+  try {
+    const {
+      builds,
+      deployments,
+      metrics: { mrCycleTime, openMergeRequestsCount },
+    } = await getBackfillData('groupToken', projectId, projectName, trackingBranch);
+
+    backfillData.builds = builds;
+    backfillData.deployments = deployments;
+    backfillData.metrics.mrCycleTime = mrCycleTime;
+    backfillData.metrics.openMergeRequestsCount = openMergeRequestsCount;
+  } catch (err) {
```

Comment:
@kbielaski This means that if **any** of the backfills fail (i.e. just deployments) then the entire dataProvider invocation will fail, right? Is this the desired behavior? Kind of worried we'll get customers with one piece of bad data who can't get dataProvider working.

Reply:
Yes, I am ok with that behavior because if we return a successful partial response from dataProvider, we can't re-enqueue part of the backfill. It is either a full fail or a full success.